### ISSUE TITLE
DB file for report - Results table has all agent entries with same agent

### ISSUE THREAD

**Issue by carlosmcconceicao** [damies13/rfswarm#79]
**Describe the bug**
Created a test plan with several test cases.
When executing I can see the manager assigning different agents for the execution.
After execution, the db file with the results table shows always the same Agent in the Agent column for all table entries. This does not match the Agent distribution shown by the master during execution.

**To Reproduce**
1. Open two Agents with different names (Agent1 and Agent2), and one Manager
2. Create test plan with 3 test cases that will have multiple iterations for the set period of 320 seconds.
3. Click on Run
4. Manager console shows the execution assignment distributed by both agents
5. When execution finishes, I open the DB file generated.
6. The results table, has all entries on column Agent with the same agent name "Agent2".

**Expected behavior**
I expected that part of the lines on this table would be associated with Agent1

**Desktop (please complete the following information):**
 - OS: Windows 10

**Comment by damies13** [comment]
Hi Carlos, 

Thanks for the report, i'm having some trouble reproducing this. I set up this scenario which should be as you described, and I have tried running it on both MacOS and Windows 10

![image](https://user-images.githubusercontent.com/15232871/112318305-9e448200-8cf8-11eb-9fed-60d9f3c61a00.png)


So in the interest of confirming this issue I've attached a a robot file and the scenario, and would appreciate it if you could run these with your agents and attach the the db file result. this test only uses the builtin library so you won't need to install any libraries to run this test.

[Issue-#79.zip](https://github.com/damies13/rfswarm/files/6197627/Issue-.79.zip)

This way we can investigate further.

**Comment by carlosmcconceicao** [comment]
Hello Dave! Thank you for your quick reply. My scenario is different, seems to have a different result, I'm sending it attached with my basic robot test and the db result from last run.

[robot-and-scenario.zip](https://github.com/damies13/rfswarm/files/6206167/robot-and-scenario.zip)

**Comment by damies13** [comment]
Hi Carlos,

while I didn't have the files:
  Resource          ../Resources/Libraries.robot
  Resource          ../Resources/performance.robot
  Resource          ../Environment/environment.robot

I simply commented them out and creates a keyword for Think_Time as it was the only keyword I seemed to be missing and was able to run your scenario, as you can see from the screen shot below I am getting results from both agents.

![image](https://user-images.githubusercontent.com/15232871/112657225-eea91480-8e9d-11eb-8447-5c9b2b781594.png)

What I found interesting from your DB file, Agent2 sends response data but doesn't do the regular status updates, Agent1 did the regular status updates but didn't send any results back, this is very weird.

Can I ask you to run the following 4 commands on both agents and send me the results for both agents as it may help me figure out what is happening.
```
rfswarm_agent -v
robot --version
python3 -V
python -V 
```

What I suspect might be happening is the agent and robot framework are running under different python versions which might explain at least some of what is happening.

Dave.

**Comment by carlosmcconceicao** [comment]
Hi, 

So, from the files I sent you, I was running Manager and two Agents in the same Windows machine, so all sharing the same versions:
_Python 3.8.3
Robot Framework 3.2.1 (Python 3.8.3 on win32)
Robot Framework Swarm: Manager
        Version 0.6.4_

Today I repeated with the agents running each one inside a docker container, and the Manager still running on the windows machine (that also hosts the docker containers), and had the same result.
versions are:
_Python 3.8.5
Robot Framework 3.2.1 (Python 3.8.5 on linux)
Robot Framework Swarm: Run Agent
        Version 0.6.4_

I'm attaching the agent INI files, 
[Agent_ini_files.zip](https://github.com/damies13/rfswarm/files/6213026/Agent_ini_files.zip)

**Comment by damies13** [comment]
Hi Carlos,

I still have no idea what's causing your issue or how to go about replicating it.

I will say that typically having multiple agents on the same machine doesn't make any sense (unless it's just for learning and trying things out) because the agent is multi-threaded and most of the load is generated by the robots which all run as seperate processes, so one agent is able to use all the system resources of the machine it's running on.

The only use case where it makes sense to run multiple agents in docker containers on the same machine is where you are driving a desktop application (Libraries like sikuli, white, autoit, flaui, etc where the only 1 copy of the desktop app can be running), running the desktop apps in a docker container would allow more robots on a physical machine.


One thing that might be happening, though I don't think so is the (non docker) agents might be tripping up because they are using the directory, I wouldn't recommend this as it's likely to cause file contention, so in your ini files I would suggest changing:
```
agentdir = /home
```
to:
```
agentdir = /home/Agent1
```
and
```
agentdir = /home/Agent2
```

Having said all that, what you are doing "should" work, though I haven't specifically tested it, i'll have a go at trying this configuration and see if I can replicate it.

Dave.

**Comment by damies13** [comment]
Hi Carlos,

I have some good news, while I still haven't been able to reproduce this issue, I have a better idea what happened now, somehow the data being stored for Agent1 got stored with the key for K3.

Let me show you what happened, this is the Metric table:
<img width="251" alt="Screen Shot 2021-03-27 at 10 54 18 am" src="https://user-images.githubusercontent.com/15232871/112705489-cabff000-8eea-11eb-9286-9c4abbc34e7e.png">

As you can see:
- Metric 2 is Agent2
- Metric 4 is Agent1
- Metric 6 is K3

Now when we look at the Metrics table:
<img width="446" alt="Screen Shot 2021-03-27 at 10 57 34 am" src="https://user-images.githubusercontent.com/15232871/112705660-8719b600-8eeb-11eb-8084-d14a0b99d771.png">

As you can see we have a "Status" of "Critical" for Metric 4 and Metric 6 but Metric 6 is not an agent, so it can't have a "Status".

Running in warning or critical state is not advised and could have caused this issue, also running in with the CPU at 100% on your agent or manager will make your test results invalid as you can't know if the slow response time is due to the AUT being stressed or you agent being stressed, likewise if the manager is stressed then it's likely to corrupt data or even loose data.

Can you rerun your tests ensuring your agents and manager are not in the warning or critical state and see if it still happens?

Meanwhile I'll have a look and see if I can make the code more robust.

Dave.

**Comment by damies13** [comment]
Hi Carlos,

After spending a bit of time reviewing the code related to registering an agent, I think I figured out what might have happened.

First a question was this the second or later run after opening the manager, i.e. did you open the manager and run a test, then run a second test with either the same or different scenario when you got this issue? if so I'm pretty sure I figured out what the issue is and the checkin above should fix it.

Dave.

**Comment by damies13** [comment]
Hi Carlos,

I haven't heard from you for almost a week, but as I believe I have found the issue and fixed I have merged this fix into the 0.6.5 branch so it will be included in that release.

You can try Version 0.6.5 of rfswarm, which will be released very soon, and if this issue continues we can reopen this issue.

Dave.
